src/main/java/ru/practicum/shareit/request/service/ItemRequestServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package ru.practicum.shareit.request.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.stereotype.Service;
+import ru.practicum.shareit.item.dao.ItemRepository;
+
+import ru.practicum.shareit.item.exception.IncorrectParameterException;
+import ru.practicum.shareit.item.mapper.ItemMapper;
+import ru.practicum.shareit.item.model.Item;
+import ru.practicum.shareit.request.dao.ItemRequestRepository;
+import ru.practicum.shareit.request.dto.ItemRequestDto;
+import ru.practicum.shareit.request.exception.IncorrectPageParametrException;
+import ru.practicum.shareit.request.exception.ItemRequestNotFoundException;
+import ru.practicum.shareit.request.mapper.ItemRequestMapper;
+import ru.practicum.shareit.request.model.ItemRequest;
+import ru.practicum.shareit.user.dao.UserRepository;
+import ru.practicum.shareit.user.exceptions.UserNotFoundException;
+import ru.practicum.shareit.user.model.User;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ItemRequestServiceImpl implements ItemRequestService {
+    private final ItemRequestRepository requestRepository;
+    private final UserRepository userRepository;
+    private final ItemRepository itemRepository;
+
+    @Override
+    public ItemRequestDto addItemRequest(Integer userId, ItemRequestDto itemRequestDto) {
+        ItemRequest itemRequest = new ItemRequest();
+        User user = getRequestorUser(userId);
+
+        itemRequest.setRequestor(user);
+
+        itemRequest.setCreated(new Date());
+        if (itemRequestDto.getDescription() == null) {
+            throw new ItemRequestNotFoundException("Описание не может быть пустым");
+        }
+        itemRequest.setDescription(itemRequestDto.getDescription());
+        requestRepository.save(itemRequest);
+
+        return ItemRequestMapper.toItemRequestDto(itemRequest);
+    }
+
+    private User getRequestorUser(Integer userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (!optionalUser.isPresent()) {
+            throw new UserNotFoundException("Пользователь не найден");
+        }
+
+        User user = optionalUser.get();
+
+        return user;
```

Comment:
Можно сразу возвращать результат get
```suggestion
        return optionalUser.get();
```

А лучше заменить все конструкцию на один метод [orElseThrow()](https://habr.com/ru/post/346782/#:~:text=%D0%B2%D1%8B%D0%B7%D1%8B%D0%B2%D0%B0%D0%B5%D1%82%20%D1%83%D0%BA%D0%B0%D0%B7%D0%B0%D0%BD%D0%BD%D1%8B%D0%B9%20%D0%BC%D0%B5%D1%82%D0%BE%D0%B4.-,orElseThrow(),-%E2%80%94%20%D0%B2%D1%8B%D0%B1%D1%80%D0%B0%D1%81%D1%8B%D0%B2%D0%B0%D0%B5%D1%82%20%D0%B8%D1%81%D0%BA%D0%BB%D1%8E%D1%87%D0%B5%D0%BD%D0%B8%D0%B5.) он выбросит ошибку, если Optional будет empty или вернет значение из optional